recipes/cassandra-cpp-driver/all/patches/2.15.3/fix-cmake.patch

```diff
@@ -0,0 +1,466 @@
+ endif()
+ 
+ #-------------------------------------
+@@ -240,62 +250,17 @@ else()
+   set(INSTALL_HEADER_DIR "include")
+ endif()
+ 
+-if(CASS_INSTALL_PKG_CONFIG)
+-  if(NOT WIN32)
+-    find_package(PkgConfig)
+-    if(PKG_CONFIG_FOUND)
+-      set(prefix ${CMAKE_INSTALL_PREFIX})
+-      set(exec_prefix ${CMAKE_INSTALL_PREFIX})
+-      set(libdir ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR})
+-      set(includedir ${CMAKE_INSTALL_PREFIX}/${INSTALL_HEADER_DIR})
+-      set(version ${PROJECT_VERSION_STRING})
+-    endif()
+-  endif()
+-endif()
+-
+ # Determine if the header should be installed
+ if(CASS_INSTALL_HEADER)
+   file(GLOB CASS_API_HEADER_FILES ${CASS_INCLUDE_DIR}/*.h)
+   install(FILES ${CASS_API_HEADER_FILES} DESTINATION ${INSTALL_HEADER_DIR})
+ endif()
+ 
+-# Install the dynamic/shared library
+-if(CASS_BUILD_SHARED)
+-  install(TARGETS cassandra
+-    RUNTIME DESTINATION ${INSTALL_DLL_EXE_DIR}  # for dll/executable/pdb files
+-    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}  # for shared library
+-    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}) # for static library
+-  if(CASS_INSTALL_PKG_CONFIG)
+-    if(NOT WIN32)
+-      if(PKG_CONFIG_FOUND)
+-        configure_file("${PROJECT_SOURCE_DIR}/packaging/cassandra.pc.in" "cassandra.pc" @ONLY)
+-        install(FILES "${CMAKE_CURRENT_BINARY_DIR}/cassandra.pc"
+-          DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
+-      endif()
+-    endif()
+-  endif()
+-  if(WIN32)
+-    install(FILES $<TARGET_PDB_FILE:cassandra>
+-      DESTINATION "${INSTALL_DLL_EXE_DIR}"
+-      OPTIONAL)
+-  endif()
+-endif()
+-
+-if(CASS_BUILD_STATIC)
+-  install(TARGETS cassandra_static
+-    RUNTIME DESTINATION ${INSTALL_DLL_EXE_DIR}  # for dll/executable/pdb files
+-    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}  # for shared library
+-    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}) # for static library
+-  if(CASS_INSTALL_PKG_CONFIG)
+-    if(NOT WIN32)
+-      if(PKG_CONFIG_FOUND)
+-        configure_file("${PROJECT_SOURCE_DIR}/packaging/cassandra_static.pc.in" "cassandra_static.pc" @ONLY)
+-        install(FILES "${CMAKE_CURRENT_BINARY_DIR}/cassandra_static.pc"
+-          DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
+-      endif()
+-    endif()
+-  endif()
+-endif()
++# Install the library
++install(TARGETS ${PROJECT_LIB_NAME_TARGET}
++  RUNTIME DESTINATION ${INSTALL_DLL_EXE_DIR}  # for dll/executable/pdb files
++  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}  # for shared library
++  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}) # for static library
+ 
+ #-------------------
+ # Uninstall target
+diff --git a/src/third_party/curl/CMakeLists.txt b/src/third_party/curl/CMakeLists.txt
+index 63f391752..8c00dd203 100644
+--- a/src/third_party/curl/CMakeLists.txt
++++ b/src/third_party/curl/CMakeLists.txt
+@@ -3,16 +3,8 @@ file(GLOB SOURCES *.cpp *.hpp)
+ source_group("Source Files\\" REGULAR_EXPRESSION ".*\\.c(pp)?")
```

Comment:
We should really use the conan curl recipe/package.

(unless this directory only contains the code to use curl instead of the curl source tree itself)

Reply:
Unfortunately curl implementation that uses Cassandra represents a private piece of code from original sources.

Reply:
Does replacing these sources with a link to `CONAN_PKG::libcurl` work?
If libcurl is built as a static library, all symbols are visible.

Reply:
@madebr But what about headers? https://github.com/curl/curl/blob/master/lib/hostcheck.h seems to be private as well. Anyway, such breaking through the public library API looks tricky to me.

Reply:
Indeed, you're right. No hostcheck.h is installed by cci's libcurl.

Can you try the following?
Add `libcurl/7.72.0` to the requirements of the test_package and see whether libcurl and cassandra cause problems.

Either way, you cannot use cci's libcurl.
But this may cause a static libcurl being imcompatible with cassandra-cpp-driver.

Reply:
Are you talking about violating of ODR for `Curl_cert_hostcheck`?

Reply:
Yes, symbol of static `libcurl.a` <-> this one.

Reply:
So, it seems that the code of cassandra driver needs to be patched, but is this a good way to solve this problem? In comparison to adding a package to CCI with some usage limitations.

Reply:
@vaerizk 
Limitations are totally fine.
We're still in the process of determining what those limitations are.
If a patch can fix the issues, maybe these can be upstreamed and others can benefit from them.

Reply:
@madebr We have tested the test_package with `cassandra-cpp-driver` and `libcurl` dependencies included at the same time. Seems it works by good fortune :smile: .

We have included the following examples, using SSL  - this one for cassandra driver, and this one for libcurl - into one executable to make sure both `Curl_cert_hostcheck`-s are invoked and preserved into final executable after linking. The package with test executable is built using `-o:shared=false` to use static variants of dependencies and `-s:build_type=Debug` to keep readable names in the output PE binary.

And now we have two symbols in the executable:
- `?Curl_cert_hostcheck@@YAHPEBD0@Z` (unmangled: `int __cdecl Curl_cert_hostcheck(char const * __ptr64,char const * __ptr64)`) - from https://github.com/datastax/cpp-driver/blob/master/src/third_party/curl/hostcheck.cpp , compiled as C++ code;
- `Curl_cert_hostcheck` - from https://github.com/curl/curl/blob/master/lib/hostcheck.c#L128 , compiled as C code.